File: src/main/java/com/uwetrottmann/trakt/v2/entities/HistoryEntry.java
package com.uwetrottmann.trakt.v2.entities;

import org.joda.time.DateTime;

public class HistoryEntry {

    public Integer id;
    public DateTime watched_at;
    public String action;
    public String type;

    public Episode episode;
    public Show show;

    public Movie movie;

}

File: src/main/kotlin/com/redissi/trakt/entities/HistoryEntry.kt
package com.redissi.trakt.entities

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import java.time.OffsetDateTime

@JsonClass(generateAdapter = true)
class HistoryEntry(
    val id: Long? = null,
    @Json(name = "watched_at")
    val watchedAt: OffsetDateTime? = null,
    val action: String? = null,
    val type: String? = null,
    val episode: Episode? = null,
    val show: Show? = null,
    val movie: Movie? = null
)